src/main/java/uk/gov/hmcts/reform/civil/aspect/EventAllowedAspect.java

```diff
@@ -51,11 +51,14 @@ public Object checkEventAllowed(
         CaseDetails caseDetails = callbackParams.getRequest().getCaseDetails();
         CaseData caseData = callbackParams.getCaseData();
         StringBuilder stateHistoryBuilder = new StringBuilder();
-        FlowState flowState = flowStateAllowedEventService.getFlowState(caseData);
-        stateFlowEngine.evaluate(caseData).getStateHistory().forEach(s -> {
-            stateHistoryBuilder.append(s.getName());
-            stateHistoryBuilder.append(", ");
-        });
+        FlowState flowState = FlowState.fromFullName("MAIN.DRAFT");
```

Comment:
This would define a draft state for all events then will overwrite for non migration events. For better clarity suggest to use -

  if (caseEvent.equals(CaseEvent.migrateCase)) {
FlowState flowState = FlowState.fromFullName("MAIN.DRAFT");
} else {
Flowstate flowState = flowStateAllowedEventService.getFlowState(caseData);
            stateFlowEngine.evaluate(caseData).getStateHistory().forEach(s -> {
                stateHistoryBuilder.append(s.getName());
                stateHistoryBuilder.append(", ");
            });
        }

Reply:
@akriti-hmcts  i tried this but because the flowState variable is used only for the logs below, it would still have to be initialized outside of the if statements. instead i moved this code block closer to the logs so we don't have to initialize it with any value.